src/linked_list.py

```diff
@@ -50,17 +51,16 @@ def pop(self):
     def search(self, val):
         """Search through list for a node and return that node."""
         current = self.head
-        found = False
         if self.length > 0:
-            while current and found is False:
+            while current:
                 if current.value == val:
-                    found = True
-                    return current
+                    break
                 else:
                     current = current.next_node
-            if current is None:
+            if current:
+                return current.value
+            else:
                 raise ValueError("Value not found in list.")
-        return None
 
     def display(self):
         """Input a linked list and return a string showing list in tuples."""
```

Comment:
You'll need to fix this method. You're calling `str` on your node instances. However, you haven't implemented a `__str__` method on your class, which means that `str(one_of_your_nodes)` will just return the default string representation of a class, which looks like: `<linked_list.Node object at 0x7f2e0d285cf8>`

So either implement a `__str__` method, or get the string representation of the node's value.